### PR TITLE
Address Sanitizer

### PR DESCRIPTION
**Pull request by eidenyoshida**

### CODE REVIEW

**Review by saadrahim — COMMENTED** [review]
Looks good, need to see results on CI

**Comment by eidenyoshida** [comment]
Debug build is supposed to run with address sanitizer enabled. Do we expect to see extra verbose output from our tests when this is enabled?

**Comment by saadrahim** [comment]
Only if it detects something. Not sure if there is a default output.

File: CMakeLists.txt [hunk 1]
```diff
@@ -53,6 +53,20 @@ set(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE CACHE BOOL "Add paths to linker searc
...
 set(BUILD_SHARED_LIBS ON CACHE BOOL "Build shared")
 
+# Set the default value of BUILD_SHARED_LIBS
+set(ADDRESS_SANITIZER OFF CACHE BOOL "Build with address sanitizer")
+
+if ( ${ADDRESS_SANITIZER} )
+    set ( HSAKMT_C_FLAGS "${HSAKMT_C_FLAGS}" -fsanitize=address )
+    set ( HSAKMT_LINK_FLAGS "${HSAKMT_LINK_FLAGS}" -fsanitize=address )
+    if ( BUILD_SHARED_LIBS} )
+        set ( HSAKMT_LINK_FLAGS "${HSAKMT_LINK_FLAGS}" -shared-libasan )
```
Comment (saadrahim) [inline CMakeLists.txt:63]: we should change the variable name. HSAKMT is a rocm project. Let's call it rocRAND or something generic.

**Review by saadrahim — CHANGES_REQUESTED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (saadrahim) [reply CMakeLists.txt:63]: Code needs to move down. we need to customize for us.

**Comment by b-sumner** [comment]
I think this looks fine (but agree that HSAKMT can be changed).  I am wondering though about a debug build without the sanitizer.  The sanitizer runtime can add a fair amount of memory overhead (in addition to the other runtime overhead)  which could be problematic in some cases.

**Comment by saadrahim** [comment]
@eidenyoshida you may want to finish this off.

**Comment by eidenyoshida** [comment]
Opened a separate PR to add sanitizer flags to install script

**Pull request closed without merging**